swap_meet/vendor.py

```diff
@@ -0,0 +1,56 @@
+class Vendor:
+    pass
+    def __init__(self, inventory = None):
+        if inventory == None:
+            self.inventory= []
+        else:
+            self.inventory = inventory
+    def add(self, add_item):
+        self.inventory.append(add_item)
+        return add_item
+    def remove(self, remove_item):
+        if remove_item not in self.inventory:
+            return False
+        else:
+            self.inventory.remove(remove_item)
+            return remove_item
+    def get_by_category(self, category):
+        item_list = []
+        for item in self.inventory:
+            if category == item.category:
+                item_list.append(item)
+        return item_list
+    def swap_items(self, vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in vendor.inventory:
+            return False
+        if my_item in self.inventory and their_item in vendor.inventory:
```

Comment:
The conditional on line 24 will catch the case where an inventory doesn't have the item, so this check isn't needed.  If the function gets to this point, each item must be in the correct inventory.